test/modules/TestAnalyzer/AggregateCloudOperationReport.cs

```diff
@@ -49,5 +49,19 @@ public class CloudOperationReport
 
             public DateTime LastReceivedAt { get; set; }
         }
+
+        bool IsPassedHelper()
+        {
+            bool noUnexpectedStatus = this.StatusCodes.All(
+                x => x.StatusCode.StartsWith("200") ||
+                x.StatusCode.Equals("OK") ||
+                (x.StatusCode.Equals("0") && this.ModuleId.Contains("DirectMethod")));
+
+            // The SDK does not allow edgehub to de-register from iothub subscriptions, which results in DirectMethod clients sometimes receiving status code 0.
+            // We expect to get this status sometimes, but if we receive too many we should fail the tests.
+            bool statusCodeZeroBelowThreshold = this.StatusCodes.Where(s => s.StatusCode.Equals("0")).Count() < (this.StatusCodes.Count / 100);
```

Comment:
Status code 0 only happens on edgehub restart.

I picked a 1% threshold for the proportion of status code 0 to other responses. The main factor that determines if this threshold can be inadvertently exceeded is the edgehub restart interval. Seeing as how the default is every 4 hours, we would need a restart interval of < 5 min to be problematic (guess).

Reply:
This change is a workaround for that git hub issue, right?  If so, please add a link pointing to that github issue.

Reply:
Why don't put the github issue link there, I think it is ok to put github issue link, but not internal link e.g. task or PBI.

Reply:
I had this before, but force pushed to resolve the issue we had on Friday. Will re-add.